comms/src/connection/message.rs

```diff
@@ -71,6 +71,24 @@ pub struct MessageEnvelopeHeader<PubKey> {
 }
 
 impl<PubKey: PublicKey> MessageEnvelopeHeader<PubKey> {
+    /// Construct a new MessageEnvelopeHeader from its member variables
+    pub fn new(
+        version: u8,
+        source: PubKey,
+        dest: NodeDestination<PubKey>,
+        signature: Vec<u8>,
```

Comment:
Shouldn't this signature be of type SchnorrSignature? I guess it doesn't matter much as it will be serialized soon

Reply:
It is currently implemented as specified in RFC-0172, maybe this could be changed in the future.